main/src/ca/uwaterloo/flix/language/phase/Verifier.scala

```diff
@@ -528,6 +565,78 @@ object Verifier {
     else failMismatchedTypes(tpe1, tpe2, loc)
   }
 
+  /**
+    * Asserts that the list of types `ts` matches the list of java classes `cs`
+    */
+  private def checkJavaParameters(ts: List[MonoType], cs: List[Class[_]], loc: SourceLocation): Unit = {
+    if (ts.length != cs.length)
+      throw InternalCompilerException("Number of types in constructor call mismatch with parameter list", loc)
+    ts.zip(cs).foreach { case (tp, klazz) => checkJavaSubtype(tp, klazz, loc) }
+  }
+
+  /**
+    * Asserts that `tpe` is a subtype of the java class type `klazz`.
+    */
+  @tailrec
+  private def checkJavaSubtype(tpe: MonoType, klazz: Class[_], loc: SourceLocation): MonoType = {
+    val monoToPrimitiveName = Map[MonoType, String](
+      MonoType.Int8 -> "byte",
+      MonoType.Int16 -> "short",
+      MonoType.Int32 -> "int",
+      MonoType.Int64 -> "long",
+      MonoType.Float32 -> "float",
+      MonoType.Float64 -> "double",
+      MonoType.Bool -> "boolean",
+      MonoType.Char -> "char",
+      MonoType.Unit -> "void"
+    )
+
+    val monoToClassName = Map[MonoType, String](
+      MonoType.String -> "java.lang.String",
```

Comment:
I stedet for Class.forName(string) kan du så ikke bruge `classOf[String]` eller noget i den stil? undgå de her typo-sensitive strenge